### PR TITLE
156577836 Written test for API endpoints

### PR DESCRIPTION
**Pull request by hariclerry**
### What does this PR do?
Have API tests on API Endpoints in Health Check App

### Description of Tasks to be complete
Have the following Test files running;
hc/api/tests/test_admin.py
hc/api/tests/test_badge.py
hc/api/tests/test_check_model.py
hc/api/tests/test_create_check.py
hc/api/tests/test_ensuretriggers.py
hc/api/tests/test_list_checks.py
hc/api/tests/test_notify.py
hc/api/tests/test_pause.py
hc/api/tests/test_ping.py
hc/api/tests/test_sendalerts.py

### How should this be tested manually?

RUN; python manage.py test hc/api/tests

### Background context;
Installed more dependencies such mock and premailer. 
### Any pivotal tracker user stories?
#156577836

### CODE REVIEW

File: hc/api/tests/test_create_check.py [hunk 1]
```diff
@@ -77,5 +83,22 @@ def test_it_rejects_non_string_name(self):
...
+        r = self.post({"api_key": "abc", "channels": "*"})
+
+        self.assertEqual(r.status_code, 201)
+        check = Check.objects.get()
+        self.assertEqual(check.channel_set.get(), channel)
+
+    # ### Test for the 'timeout is too small' and 'timeout is too large' errors
+    def test_it_rejects_small_timeout(self):
+        self.post({"api_key": "abc", "timeout": 0},
+                  expected_error="timeout is too small")
```
Comment (aenkya) [inline hc/api/tests/test_create_check.py]: how do you know if this test has passed

File: hc/api/tests/test_sendalerts.py [hunk 2]
```diff
@@ -37,6 +38,6 @@ def test_it_handles_grace_period(self):
         check.save()
 
         # Expect no exceptions--
-        Command().handle_one(check)
+        self.assertTrue(Command().handle_one(check))
```
Comment (aenkya) [inline hc/api/tests/test_sendalerts.py]: There was no need to modify this one. The command would return an exception if there was an issue. So the test is by default passing if the command does not return an exception. That's what the comment above was saying

File: hc/api/tests/test_ping.py [hunk 3]
```diff
@@ -63,11 +66,37 @@ def test_it_reads_forwarded_protocol(self):
...
+        self.assertEqual(self.check.status, "up")
     ### Test that a post to a ping works
+    def test_post_works(self):
+        csrf_client = Client(enforce_csrf_checks=True)
+        r = csrf_client.post("/ping/%s/" %self.check.code, "hello world",
+                             content_type="text/plain")
+        self.assertEqual(r.status_code, 200)
+        ping = Ping.objects.latest("id")
+        self.assertEqual(ping.method, "POST")
+        # self.assertEqual(ping.body, "hello world")
```
Comment (aenkya) [inline hc/api/tests/test_ping.py]: take out comments for unused code

File: hc/api/tests/test_ping.py [hunk 4]
```diff
@@ -63,11 +66,37 @@ def test_it_reads_forwarded_protocol(self):
...
+        self.check.status = "paused"
+        self.check.save()
+
+        r = self.client.get("/ping/%s/" % self.check.code)
+        self.assertEqual(r.status_code, 200)
+
+        self.check.refresh_from_db()
+        self.assertEqual(self.check.status, "up")
     ### Test that a post to a ping works
+    def test_post_works(self):
```
Comment (aenkya) [inline hc/api/tests/test_ping.py]: `test_post_ping_works` might be a better name.
`test_post_works` is a name that can be applied to testing the `post` method in general in which case it feels like you are testing the REST protocol

File: hc/api/tests/test_create_check.py [hunk 5]
```diff
@@ -77,5 +83,22 @@ def test_it_rejects_non_string_name(self):
...
+
+    def test_it_rejects_large_timeout(self):
```
Comment (aenkya) [inline hc/api/tests/test_create_check.py]: I don't think this test is complete either

**Review by aenkya — CHANGES_REQUESTED** [review]

File: hc/api/tests/test_create_check.py [hunk 6]
```diff
@@ -48,22 +51,25 @@ def test_it_works(self):
...
 
     def test_it_handles_missing_request_body(self):
-        ### Make the post request with a missing body and get the response
-        r = {'status_code': 400, 'error': "wrong api_key"} ### This is just a placeholder variable
-        self.assertEqual(r['status_code'], 400)
-        self.assertEqual(r["error"], "wrong api_key")
+        # Make the post request with a missing body and get the response
+        r = self.client.post(self.URL, content_type="application/json")
+        self.assertEqual(r.status_code, 400)
+        self.assertEqual(r.json()["error"], "wrong api_key")
```
Comment (dondrzzy) [inline hc/api/tests/test_create_check.py:65]: For now i think we can do with many asserts but asserting 400 should be sufficient to show that it handles a missing body. i.e. bad request.

File: hc/api/tests/test_create_check.py (hunk 1 above)
Comment (dondrzzy) [reply hc/api/tests/test_create_check.py]: Does self.post(object) actually post?
 assuming it does without specifying a URL, maybe its posting to the same url its exists in.

File: hc/api/tests/test_ping.py (hunk 4 above)
Comment (dondrzzy) [reply hc/api/tests/test_ping.py]: Yeah, you're right George

**Review by dondrzzy — COMMENTED** [review]
There are a few comments in the files. Please look through and make the necessary changes.

File: hc/api/tests/test_sendalerts.py (hunk 2 above)
Comment (hariclerry) [reply hc/api/tests/test_sendalerts.py]: Okay.  I also noticed it and have resolved it. Thanks for the feedback on that.

File: hc/api/tests/test_ping.py (hunk 3 above)
Comment (hariclerry) [reply hc/api/tests/test_ping.py]: Okay. Worked on it.

File: hc/api/tests/test_ping.py (hunk 4 above)
Comment (hariclerry) [reply hc/api/tests/test_ping.py]: Worked on it.

File: hc/api/tests/test_create_check.py (hunk 5 above)
Comment (hariclerry) [reply hc/api/tests/test_create_check.py]: Yes, it is incomplete. I have fixed it. Thanks for pointing that out.

File: hc/api/tests/test_create_check.py (hunk 1 above)
Comment (hariclerry) [reply hc/api/tests/test_create_check.py]: I fixed it by adding the assert method.

File: hc/api/tests/test_create_check.py (hunk 6 above)
Comment (hariclerry) [reply hc/api/tests/test_create_check.py:65]: @dondrzzy, I agree with you. I am working on it.

**Comment by King-Benx** [comment]
please ensure that you maintain pep8 standards while working

**Review by King-Benx — CHANGES_REQUESTED** [review]
Please ensure that you maintain pep8 standards

**Comment by hariclerry** [comment]
@King-Benx, thanks for pointing that out. I have implemented it.

**Pull request merged**